x-pack/plugin/vector-tile/src/main/java/org/elasticsearch/xpack/vectortile/feature/FeatureFactoryUtils.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.vectortile.feature;
+
+import org.elasticsearch.geometry.Rectangle;
+import org.locationtech.jts.geom.Envelope;
+
+/**
+ * Utility functions to transforms WGS84 coordinates into spherical mercator.
+ */
+class FeatureFactoryUtils {
+
+    private FeatureFactoryUtils() {
+        // no instances
+    }
+
+    /**
+     * Gets the JTS envelope for z/x/y/ tile in spherical mercator projection.
+     */
+    public static Envelope getJTSTileBounds(int z, int x, int y) {
+        return new Envelope(getLong(x, z), getLong(x + 1, z), getLat(y, z), getLat(y + 1, z));
+    }
+
+    /**
+     * Gets the {@link org.elasticsearch.geometry.Geometry} envelope for z/x/y/ tile
+     * in spherical mercator projection.
+     */
+    public static Rectangle getTileBounds(int z, int x, int y) {
+        return new Rectangle(getLong(x, z), getLong(x + 1, z), getLat(y, z), getLat(y + 1, z));
+    }
+
+    private static double getLong(int x, int zoom) {
+        return lonToSphericalMercator(x / Math.pow(2, zoom) * 360 - 180);
```

Comment:
Ss this code performance-sensitive? If so we might want to do leverage `Math#scalb` instead:
```suggestion
        return lonToSphericalMercator(Math.scalb(x, -zoom) * 360 - 180);
```